drivers/input/input_cap12xx.c

```diff
@@ -0,0 +1,232 @@
+	int r;
+
+	if (!device_is_ready(config->i2c.bus)) {
+		LOG_ERR("I2C controller device not ready");
+		return -ENODEV;
+	}
+
+	data->dev = dev;
+
+	k_work_init(&data->work, cap12xx_work_handler);
+
+	if (config->int_gpio.port != NULL) {
+		if (!gpio_is_ready_dt(&config->int_gpio)) {
+			LOG_ERR("Interrupt GPIO controller device not ready");
+			return -ENODEV;
+		}
+
+		r = gpio_pin_configure_dt(&config->int_gpio, GPIO_INPUT);
+		if (r < 0) {
+			LOG_ERR("Could not confighure interrupt GPIO pin");
+			return r;
+		}
+
+		r = gpio_pin_interrupt_configure_dt(&config->int_gpio, GPIO_INT_EDGE_TO_ACTIVE);
+		if (r < 0) {
+			LOG_ERR("Could not configure interrupt GPIO interrupt");
+			return r;
+		}
+
+		gpio_init_callback(&data->int_gpio_cb, cap12xx_isr_handler,
+				   BIT(config->int_gpio.pin));
+
+		r = gpio_add_callback(config->int_gpio.port, &data->int_gpio_cb);
+		if (r < 0) {
+			LOG_ERR("Could not set gpio callback");
+			return r;
+		}
+
+		r = cap12xx_clear_interrupt(&config->i2c);
+		if (r < 0) {
+			LOG_ERR("Could not clear interrupt");
+			return r;
+		}
+
+		r = cap12xx_enable_interrupt(&config->i2c, true);
+		if (r < 0) {
+			LOG_ERR("Could not configure interrupt");
+			return r;
+		}
+	}
+#ifdef CONFIG_INPUT_CAP12XX_POLL
+	else {
+		k_timer_init(&data->timer, cap12xx_timer_handler, NULL);
+
+		r = cap12xx_enable_interrupt(&config->i2c, false);
+		if (r < 0) {
+			LOG_ERR("Could not configure interrupt");
+			return r;
+		}
+
+		k_timer_start(&data->timer, K_MSEC(CONFIG_INPUT_CAP12XX_PERIOD),
+			      K_MSEC(CONFIG_INPUT_CAP12XX_PERIOD));
+	}
+#endif
+	if (config->repeat == 0) {
+		r = i2c_reg_write_byte_dt(&config->i2c, REG_REPEAT_ENABLE, REPEAT_DISABLE);
+		if (r < 0) {
+			LOG_ERR("Could not disable repeated events");
+			return r;
+		}
+		LOG_DBG("disabled repeated events");
+	} else {
+		r = i2c_reg_write_byte_dt(&config->i2c, REG_REPEAT_ENABLE, REPEAT_ENABLE);
+		if (r < 0) {
+			LOG_ERR("Could not enable repeated events");
+			return r;
+		}
+		LOG_DBG("enabled repeated events");
+	}
+
+	return 0;
+}
+
+#define CAP12XX_INIT(index)                                                                        \
+	BUILD_ASSERT(DT_INST_PROP_LEN(index, input_codes) == DT_INST_PROP(index, channels));       \
+	static const uint8_t cap12xx_input_channels = DT_INST_PROP(index, channels);               \
+	static const uint16_t cap12xx_input_codes_##inst[] = DT_INST_PROP(index, input_codes);     \
+	static const struct cap12xx_config cap12xx_config_##index = {                              \
+		.i2c = I2C_DT_SPEC_INST_GET(index),                                                \
+		.int_gpio = GPIO_DT_SPEC_INST_GET_OR(index, int_gpios, {0}),                       \
+		.input_channels = cap12xx_input_channels,                                          \
+		.input_codes = cap12xx_input_codes_##inst,                                         \
+		.repeat = DT_INST_PROP_OR(index, repeat, 0),                                       \
```

Comment:
booleans don't need the `_OR` variants, just `DT_INST_PROP`

Reply:
Then I need to set a default in the device tree.
(Need to look up how to do that)

Reply:
you shouldn't have to, the boolean defaults to false, just change to `DT_INST_PROP` and it'll work as before